planner/mapper/targetable.go

```diff
@@ -86,6 +86,51 @@ func NewFilter() *Filter {
 	}
 }
 
+func (f *Filter) ToMap(mapping *core.DocumentMapping) map[string]any {
+	return filterObjectToMap(mapping, f.Conditions)
+}
+
+func filterObjectToMap(mapping *core.DocumentMapping, obj map[connor.FilterKey]any) map[string]any {
+	outmap := make(map[string]any)
+	if obj == nil {
+		return nil
+	}
+	for k, v := range obj {
+		switch keyType := k.(type) {
+		case *PropertyIndex:
+			subObj := v.(map[connor.FilterKey]any)
+			outkey, _ := mapping.TryToFindNameFromIndex(keyType.Index)
+			if childMapping, ok := tryGetChildMapping(mapping, keyType.Index); !ok {
```

Comment:
suggestion: The `!ok` is a bit weird here as `childMapping` is consumed in the `else` not the `if`.  I was going to suggest just inverting, but you can probably go one better and remove the duplicate `filterObjectToMap` func call and `outmap` set:

```
var propMapping *core.DocumentMapping
if childMapping, ok := tryGetChildMapping(mapping, keyType.Index); ok {
    propMapping = childMapping
} else {
    propMapping = mapping
}
outmap[outkey] = filterObjectToMap(propMapping, subObj)
```

Reply:
just going to invert the `if/else` - again I try not to be too clever to reduce lines by pulling out common stuff in `if` blocks, unless it really adds clarity/semantics somehow. Given the one-liner of each, I like the clarity it current has.

Reply:
Fair enough, happy with just the inversion - I think which one I'd go with myself would be dependent on time-of-day/position-of-Jupiter/etc lol